src/base.jl

```diff
@@ -80,22 +92,39 @@ function System(buses::Vector{Bus},
         push!(arrays, services)
     end
 
+    error_detected = false
+
     for component in Iterators.flatten(arrays)
-        add_component!(sys, component)
+        try
+            add_component!(sys, component)
+        catch
```

Comment:
I recommend checking the type of exception thrown.  If it's the error you throw for invalid data then continue.  Otherwise, rethrow.  The problem here is that if someone introduces a bug in add_component that causes an exception, that fact will be hidden and debug will be difficult.